### PR TITLE
Make warming GPS configurable/optional

### PR DESCRIPTION
**Pull request by ekigamba**
1. Add isWarmGPS() and setWarmGps(boolean) API to IKujakuMapView
2. Add mapbox_warmGps property to KujakuMapView xml so that warming can be disabled in the styling xml
3. Make warming GPS enabled by default

Todo:

- [x] Check location services and location enabled whenever **My Location Button** is clicked
- [x] Test with GPS

### CODE REVIEW

File: library/src/main/java/io/ona/kujaku/views/KujakuMapView.java [hunk 1]
```diff
@@ -186,6 +195,8 @@ private void init(@Nullable AttributeSet attributeSet) {
             @Override
             public void onClick(View v) {
                 focusOnUserLocation(true);
+                hasAlreadyRequestedEnableLocation = false;
```
Comment (vincent-karuri) [inline library/src/main/java/io/ona/kujaku/views/KujakuMapView.java]: Why is this necessary?

File: library/src/main/java/io/ona/kujaku/views/KujakuMapView.java [hunk 2]
```diff
@@ -880,12 +899,60 @@ public void onResume() {
...
+                                hasAlreadyRequestedEnableLocation = true;
+
+                                try {
+                                    // Show the dialog by calling startResolutionForResult(), and check the result
+                                    // in onActivityResult().
+                                    status.startResolutionForResult(activity, Constants.RequestCode.LOCATION_SETTINGS);
+                                } catch (IntentSender.SendIntentException e) {
+                                    Log.i(TAG, "PendingIntent unable to execute request.");
+                                }
+                            } else {
```
Comment (vincent-karuri) [inline library/src/main/java/io/ona/kujaku/views/KujakuMapView.java]: What does is the intention of this else block?

Could you add comments to better explain the high level objective? It's not that clear.

File: library/src/main/java/io/ona/kujaku/views/KujakuMapView.java (hunk 1 above)
Comment (ekigamba) [reply library/src/main/java/io/ona/kujaku/views/KujakuMapView.java]: KujakuMapView **has already requested to enable location**
